### PR TITLE
Bg monte carlo

### PR DESCRIPTION
**Pull request by yfv2ev**
This branch adds the files bgCutflow, bgAnalysis, and bg_filetests, as well as a new function in cutflow.py and a minor change in ffschema.py. bgCutflow and bgAnalysis look at the background samples, mainly through using cutflows and histograms respectively, although there is some overlap. bg_filestests was intended to test individual files, and ended up being the main tool for finding the error in some of the QCD ntuples. The new function in cutflow.py still needs some cleanup, but it creates a table of multiple cutflows, taking a list as an argument. The change in ffschema.py resolves an error in the histogram "lj_pfiso" and in one of the QCD samples. It is marked in the file in case it causes issues down the line. The most important outstanding issue is that **Dask does not pick up on this fix**, so running the QCD file in question or using the histogram mentioned above under the Dask executor can cause the programs to crash.

### CODE REVIEW

File: sidm/studies/Background Sample List (Reference).txt
Comment (btcardwell) [inline sidm/studies/Background Sample List (Reference).txt:1]: Hi @yfv2ev, is this file accessed anywhere in the code, or is it just for your visual reference? If the latter, would you mind removing it from the PR?

File: sidm/tools/histogram.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def __init__(self, axes, storage="weight", evt_mask=None):
         # Allow all events to pass if no mask is explicitly provided
         self.evt_mask = (lambda objs: slice(None)) if evt_mask is None else evt_mask
         self.hist = None
-
+
```
Comment (btcardwell) [inline sidm/tools/histogram.py:23]: Hi @yfv2ev, would you mind removing this extra whitespace?

File: sidm/tools/scaleout.py [hunk 2]
```diff
@@ -7,6 +7,7 @@ def make_dask_client(address):
     """Create dask client that includes dependency installer"""
     dependencies = [
         "git+https://github.com/btcardwell/SIDM.git",
+        #"git+https://github.com/yfv2ev/SIDM.git#bgMonteCarlo"
```
Comment (btcardwell) [inline sidm/tools/scaleout.py:10]: Hi @yfv2ev, would you mind removing this line? It totally makes sense what you're up to, but I don't think we need to add it to the main branch

File: sidm/tools/sidm_processor.py [hunk 3]
```diff
@@ -143,7 +143,6 @@ def build_lepton_jets(self, objs, lj_reco):
         # take lepton jets directly from ntuples
         if lj_reco == 0:
             ljs = objs["ntuple_ljs"]
-
```
Comment (btcardwell) [inline sidm/tools/sidm_processor.py:146]: Hi @yfv2ev, could you add this blank line back in?

File: sidm/tools/ffschema.py [hunk 4]
```diff
@@ -187,7 +187,8 @@ def _build_collections(self, branch_forms):
             # handle non-vector objects
             if obj not in branch_forms:
                 branch_forms[obj] = zip_forms(
-                    {a: branch_forms.pop(f"{obj}_{a}") for a in attributes if a != "n"},
+                    #{a: branch_forms.pop(f"{obj}_{a}") for a in attributes if a != "n"}, #Original line ------------------------------ Changed here
```
Comment (btcardwell) [inline sidm/tools/ffschema.py:190]: Hi @yfv2ev, I think you can remove this commented line (your fix is probably good to keep from now on)

File: sidm/studies/processor.coffea
Comment (btcardwell) [inline sidm/studies/processor.coffea:1]: Hi @yfv2ev, would you mind removing this file from the PR?

File: sidm/studies/leptonDeltaR.ipynb
Comment (btcardwell) [inline sidm/studies/leptonDeltaR.ipynb:1]: Hi @yfv2ev, it looks like a few notebooks (like this one) were added to the PR by mistake. Unless there are changes you want to include, I think these notebooks should be removed from the PR.

File: sidm/studies/Background Sample List (Reference).txt
Comment (yfv2ev) [reply sidm/studies/Background Sample List (Reference).txt:1]: Removed it!

File: sidm/tools/histogram.py (hunk 1 above)
Comment (yfv2ev) [reply sidm/tools/histogram.py]: Should be fixed now

File: sidm/tools/scaleout.py (hunk 2 above)
Comment (yfv2ev) [reply sidm/tools/scaleout.py]: Fixed

File: sidm/tools/sidm_processor.py (hunk 3 above)
Comment (yfv2ev) [reply sidm/tools/sidm_processor.py]: Added it

File: sidm/tools/ffschema.py (hunk 4 above)
Comment (yfv2ev) [reply sidm/tools/ffschema.py]: Fixed

File: sidm/studies/processor.coffea
Comment (yfv2ev) [reply sidm/studies/processor.coffea:1]: It's deleted. Not sure what it was

File: sidm/studies/leptonDeltaR.ipynb
Comment (yfv2ev) [reply sidm/studies/leptonDeltaR.ipynb:1]: Most of them are removed, but it seems like the two that are left (darkPhotonPtEfficiency and leptonDeltaR) are exclusive to this branch and were probably removed from yours at some point. Everything else should be fixed though.

Comment (btcardwell) [reply sidm/studies/leptonDeltaR.ipynb:1]: great, thank you! I'll probably fix this in a bonus commit after merging your PR. I'll double check everything else now...

**Comment by btcardwell** [comment]
ok, all looks good @yfv2ev ! I'll squash and merge these into main now (rebase is too complicated for this one), and then I'll push another commit to fix the accidentally re-added notebooks.

**Pull request merged**